ubirch/ubirch_protocol.c

```diff
@@ -0,0 +1,179 @@
+/*!
+ * @file
+ * @brief ubirch protocol API
+ *
+ * @author Roxana Meixner
+ * @date   2019-09-10
+ *
+ * @copyright &copy; 2019 ubirch GmbH (https://ubirch.com)
+ *
+ * ```
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * ```
+ */
+
+#include "ubirch_protocol.h"
+#include "ubirch_protocol_kex.h"
+
+/**
+ * Start a new message. Writes the header data to ubirch protocol package.
+ *
+ * @param upp the ubirch protocol context
+ * @param variant protocol variant
+ * @param payload_type the payload data type indicator (0 - binary)
+ * @return -1 if upp is NULL
+ * @return -2 if the protocol version is not supported
+ */
+static int8_t ubirch_protocol_start(ubirch_protocol *upp, ubirch_protocol_variant variant) {
```

Comment:
I am wondering why the structure has changed here from inline code in the header to a static function in the .c file.
Some parts are still in the header, these are not, why?

Reply:
I've put the functions that are not part of the public API as static in the .c file for access control